.github/workflows/build_workflow.yml

```diff
@@ -0,0 +1,340 @@
+name: GP2GP Build Workflow
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  checkstyle:
+    name: Checkstyle
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
```

Comment:
maybe not in this PR but as a next iteration it would be good to consider including Gradle caching mechanisms which should improve the performance even better. Here is some analysis about the positive caching impact: https://docs.gradle.org/current/userguide/build_cache_performance.html#analyzing_performance_in_build_scans

Reply:
Nice spot, agreed that this should be done at a later phase! I will take note :)